compiler/rustc_resolve/src/late/lifetimes.rs

```diff
@@ -2557,6 +2557,10 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
                 GenericArg::Const(ct) => {
                     self.visit_anon_const(&ct.value);
                 }
+                GenericArg::Infer(inf) => {
+                    self.visit_id(inf.hir_id);
+                    i += inf.kind.is_type() as usize;
```

Comment:
🤣 Please do `if inf.kind.is_type() { i += 1; }`. I appreciate the terseness, but this will trip up ppl in the future